packages/react/src/SignatureInput/SignatureInput.tsx

```diff
@@ -0,0 +1,68 @@
+import { CloseButton, Paper, PaperProps } from '@mantine/core';
+import { createReference, HTTP_HL7_ORG, ProfileResource } from '@medplum/core';
+import { Signature } from '@medplum/fhirtypes';
+import { useMedplum } from '@medplum/react-hooks';
+import { JSX, useEffect, useRef } from 'react';
+import SignaturePad from 'signature_pad';
+
+export interface SignatureInputProps extends PaperProps {
+  readonly width?: number;
+  readonly height?: number;
+  readonly defaultValue?: Signature;
+  readonly onChange: ((value: Signature | undefined) => void) | undefined;
+}
+
+export function SignatureInput(props: SignatureInputProps): JSX.Element {
+  const medplum = useMedplum();
+  const { width = 500, height = 200, defaultValue, onChange, ...rest } = props;
+  const canvasRef = useRef<HTMLCanvasElement>(null);
+  const signaturePadRef = useRef<SignaturePad>(null);
+
+  const onChangeRef = useRef(onChange);
+  onChangeRef.current = onChange;
+
+  useEffect(() => {
+    function handleEndStroke(): void {
+      onChangeRef.current?.({
+        type: [
+          {
+            system: HTTP_HL7_ORG + '/fhir/signature-type',
+            code: 'ProofOfOrigin',
+            display: 'Proof of Origin',
+          },
+        ],
+        when: new Date().toISOString(),
+        who: createReference(medplum.getProfile() as ProfileResource),
```

Comment:
Recommend adding a prop for this. We have customers where a caregiver / practitioner is filling out a Questionnaire on behalf of a patient and then hands the device to them for signature. So the logged-in user may not be the same as the person actually doing the signing.